src/test/java/api/support/fixtures/NoticeMatchers.java

```diff
@@ -124,6 +125,9 @@ private static String findItemRepresentationCopyNumbers(InventoryItemResource it
     } else {
       int renewalLimit = getIntegerProperty(renewalsPolicy, "numberAllowed", 0);
       int renewalsRemaining = renewalLimit - renewalsCount;
+      if (renewalsRemaining < 0) {
```

Comment:
This code is starting to look very similar to the implementation in `NoticeContextUtil`, which makes me wonder if it is trying to be too general.

Branching during checks within tests suggests to me that the check is accepting the outcome of two distinct scenarios, irrespective of which scenario the current test is exercising.

I think this can obscure what the test is trying to express, possibly making it hard to understand and less useful for identifying failures.

In this case the impact is likely minor, my concern is for the cumulative effect of this in this area. Maybe addressing this is out of the scope of this work, my concern is that it may be deferred continually.

Reply:
Agree, it makes sense to me. In the last commit, I tried to make this verification less general and removed logic from this method.

Reply:
Thanks for starting to improve this area :-)